lib/convection/control/cloud.rb

```diff
@@ -20,45 +20,105 @@ def deck
         @cloudfile.deck
       end
 
-      def converge(to_stack, &block)
-        unless to_stack.nil? || stacks.include?(to_stack)
-          block.call(Model::Event.new(:error, "Stack #{ to_stack } is not defined", :error)) if block
-          return
-        end
+      def stack_groups
+        @cloudfile.stack_groups
+      end
 
+      def create_structure(list)
+        hash = {}
         deck.each do |stack|
-          block.call(Model::Event.new(:converge, "Stack #{ stack.name }", :info)) if block
-          stack.apply(&block)
-
-          if stack.error?
-            block.call(Model::Event.new(:error, "Error converging stack #{ stack.name }", :error), stack.errors) if block
-            break
-          end
+          hash[stack.name] = stack if list.include?(stack.name.to_s)
+        end
+        hash
+      end
 
+      def simple_converge(to_stack, &block)
+        deck.each do |stack|
+          break unless apply_converge(stack, &block)
           ## Stop on converge error
           break unless stack.success?
-
           ## Stop here
           break if !to_stack.nil? && stack.name == to_stack
           sleep rand @cloudfile.splay || 2
         end
       end
 
-      def diff(to_stack, &block)
-        @cloudfile.deck.each do |stack|
-          block.call(Model::Event.new(:compare, "Compare local state of stack #{ stack.name } (#{ stack.cloud_name }) with remote template", :info))
-
-          difference = stack.diff
-          if difference.empty?
-            difference << Model::Event.new(:unchanged, "Stack #{ stack.cloud_name } Has no changes", :info)
-          end
+      def list_converge(hash, list, &block)
+        list.each do |stack|
+          break unless apply_converge(hash[stack], &block)
+          sleep rand @cloudfile.splay || 2
+        end
+      end
 
-          difference.each { |diff| block.call(diff) }
+      def apply_converge(stack, &block)
+        block.call(Model::Event.new(:converge, "Stack #{ stack.name }", :info)) if block
+        stack.apply(&block)
+        if stack.error?
+          block.call(Model::Event.new(:error, "Error converging stack #{ stack.name }", :error), stack.errors) if block
+          return false
```

Comment:
@bcall-r7 pretty sure `stack.success?` below will return false anyways, in which case this isn't needed since we don't have any other logic below.